sources/PTYSession.m

```diff
@@ -3493,9 +3495,11 @@ - (void)scheduleUpdateIn:(NSTimeInterval)timeout {
 
 - (void)doAntiIdle {
     NSTimeInterval now = [NSDate timeIntervalSinceReferenceDate];
-    if (now >= _lastInput + 60) {
+    // _antiIdlePeriod-0.1 is to prevent missing every other cycle due to the timer calling us at the exact same period as we would check here.
+    // FIXME: corner case: _antiIdlePeriod>=1.1 is part 2of2 of a kludgey way to prevent a flood when period is set to 1.
+    if (now >= _lastOutput + _antiIdlePeriod - 0.1 && _antiIdlePeriod >= 1.1) {
         [_shell writeTask:[NSData dataWithBytes:&_antiIdleCode length:1]];
-        _lastInput = now;
+        _lastOutput = now;
```

Comment:
I think `_lastInput` was correct. `_lastInput` tracks the last keystroke and `_lastOutput` tracks the last time stuff was received to display on the screen, escape sequences, etc. Sending the anti-idle code acts like a keystroke for the purposes of rate limiting the anti-idle code.